src/compiler/core.ts

```diff
@@ -633,6 +633,10 @@ module ts {
         }
     }
 
+    export function getDefaultLibraryFilename(options: CompilerOptions): string {
+        return options.target === ScriptTarget.ES6 ? "lib.es6.d.ts" : "lib.d.ts";
+    }
+

```

Comment:
We use some stuff like this in the harness - @yuit was questioning the logic a week ago. Not sure if the problem persists.